omod/src/main/java/org/openmrs/module/coreapps/page/controller/providermanagement/EditProviderPageController.java

```diff
@@ -0,0 +1,239 @@
+package org.openmrs.module.coreapps.page.controller.providermanagement;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.ProviderAttribute;
+import org.openmrs.ProviderAttributeType;
+import org.openmrs.Relationship;
+import org.openmrs.RelationshipType;
+import org.openmrs.api.APIException;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.ProviderService;
+import org.openmrs.api.context.Context;
+import org.openmrs.messagesource.MessageSourceService;
+import org.openmrs.module.emrapi.account.AccountDomainWrapper;
+import org.openmrs.module.emrapi.account.AccountService;
+import org.openmrs.module.emrapi.account.AccountValidator;
+import org.openmrs.module.providermanagement.Provider;
+import org.openmrs.module.providermanagement.ProviderRole;
+import org.openmrs.module.providermanagement.api.ProviderManagementService;
+import org.openmrs.module.providermanagement.exception.InvalidRelationshipTypeException;
+import org.openmrs.module.providermanagement.exception.PersonIsNotProviderException;
+import org.openmrs.module.providermanagement.exception.SuggestionEvaluationException;
+import org.openmrs.module.uicommons.UiCommonsConstants;
+import org.openmrs.ui.framework.annotation.BindParams;
+import org.openmrs.ui.framework.annotation.MethodParam;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.page.PageModel;
+import org.springframework.context.MessageSource;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class EditProviderPageController {
+
+    protected final Log log = LogFactory.getLog(getClass());
+
+    class ProviderPatientRelationship {
+        Patient patient = null;
+        Relationship relationship= null;
+        RelationshipType relationshipType = null;
+
+        public ProviderPatientRelationship() {
+        }
+
+        public ProviderPatientRelationship(Patient patient, Relationship relationship, RelationshipType relationshipType) {
+            this.patient = patient;
+            this.relationship = relationship;
+            this.relationshipType = relationshipType;
+        }
+    }
+
+    public AccountDomainWrapper getAccount(@RequestParam(value = "personId", required = false) Person person,
+                                           @SpringBean("accountService") AccountService accountService) {
+
+        AccountDomainWrapper account;
```

Comment:
It seems  a little odd here to use the AccountDomainWrapper if this UI is only for handling the Provider/Person aspect, not the User aspect, but if there is functionality the AccountDomainWrapper provides that is helpful in this context, seems like it is fine.